src/world_model/nav2_costmap_world_model/src/costmap_world_model.cpp

```diff
@@ -27,32 +28,78 @@ namespace nav2_costmap_world_model
 CostmapWorldModel::CostmapWorldModel(const string & name)
 : Node(name + "_Node")
 {
-  costmap_ = std::make_unique<nav2_util::Costmap>(this);
+  // Create layered costmap with static and inflation layer
+  layered_costmap_ = new costmap_2d::LayeredCostmap("frame", false, false);
+  addLayer<costmap_2d::StaticLayer>("static");
+  addLayer<costmap_2d::InflationLayer>("inflation");
+  // TODO(bpwilcox): replace manual footprint to layered_costmap with parameter or from nav2_robot
+  setFootprint(0, 0);
+  layered_costmap_->updateMap(0, 0, 0);
+
 
   auto costmap_service_callback = [this](
-    const std::shared_ptr<rmw_request_id_t>/*request_header*/,
+    const std::shared_ptr<rmw_request_id_t> request_header,
     const std::shared_ptr<nav2_world_model_msgs::srv::GetCostmap::Request> request,
     const std::shared_ptr<nav2_world_model_msgs::srv::GetCostmap::Response> response) -> void
     {
       RCLCPP_INFO(
         this->get_logger(), "CostmapWorldModel::CostmapWorldModel:Incoming costmap request");
-      response->map = costmap_->getCostmap(request->specs);
+      costmap_callback(request_header, request, response);
     };
 
   // Create a service that will use the callback function to handle requests.
   costmapServer_ = create_service<nav2_world_model_msgs::srv::GetCostmap>(name + "_GetCostmap",
       costmap_service_callback);
```

Comment:
Why not simply register costmap_callback directly rather than a lambda function (costmap_service_callback) that immediately calls costmap_callback?